### ISSUE TITLE
EIN does not display plot images.

### ISSUE THREAD

**Issue by gahag** [millejoh/emacs-ipython-notebook#524]
## Problem description
EIN does not display plot images.

## Steps to reproduce the problem
```python
import matplotlib.pyplot as plt
plt.plot([1,2,3,4,6])
plt.ylabel('some numbers')
plt.show()
```

When I press C-c C-c the first time, the text output is given:
`<Figure size 640x480 with 1 Axes>`
The subsequent executions result in a dot. No image is displayed.

## System info:

```cl
("EIN system info"
 :emacs-version
 "GNU Emacs 26.2 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.8)
 of 2019-04-12"
 :emacs-bzr-version nil
 :window-system x
 :emacs-variant nil
 :os
 (:uname
  "Linux arch-blade 4.19.37-1-lts #1 SMP Sat Apr 27 10:16:32 CEST 2019 x86_64 GNU/Linux
"
  :lsb-release
  "LSB Version:	1.4
Distributor ID:	Arch
Description:	Arch Linux
Release:	rolling
Codename:	n/a
")
 :notebook
 "Name: notebook
Version: 5.7.8
Summary: A web-based notebook environment for interactive computing
Home-page: http://jupyter.org
Author: Jupyter Development Team
Author-email: jupyter@googlegroups.com
License: BSD
Location: /usr/lib/python3.7/site-packages
Requires: jinja2, tornado, pyzmq, ipython-genutils, traitlets, jupyter-core, jupyter-client, nbformat, nbconvert, ipykernel, Send2Trash, terminado, prometheus-client
Required-by: widgetsnbextension
"
 :ipython "7.3.0
"
 :image-types
 (svg imagemagick png gif tiff jpeg xpm xbm pbm postscript)
 :image-types-available
 (svg imagemagick png gif tiff jpeg xpm xbm pbm postscript)
 :request
 (:backend curl)
 :ein
 (:version "20190430.1340"
           :source-dir "/home/gahag/.emacs.d/elpa/ein-20190430.1340/")
 :lib
 ((:name "websocket"
         :path "~/.emacs.d/elpa/websocket-20190408.152/websocket.elc"
         :featurep t
         :version-var websocket-version
         :version "1.10")
  (:name "auto-complete"
         :path "~/.emacs.d/elpa/auto-complete-20170125.245/auto-complete.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "request"
         :path "~/.emacs.d/elpa/request-20181129.1138/request.elc"
         :featurep t
         :version-var request-version
         :version "0.3.0")
  (:name "deferred"
         :path "~/.emacs.d/elpa/deferred-20170901.1330/deferred.elc"
         :featurep t
         :version-var deferred:version
         :version "0.5.0")
  (:name "polymode"
         :path "~/.emacs.d/elpa/polymode-20190426.1729/polymode.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "markdown-mode"
         :path "~/.emacs.d/elpa/markdown-mode-20190305.319/markdown-mode.elc"
         :featurep t
         :version-var markdown-mode-version
         :version "2.4-dev")
  (:name "dash"
         :path "~/.emacs.d/elpa/dash-20190424.1804/dash.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "s"
         :path "~/.emacs.d/elpa/s-20180406.808/s.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "skewer-mode"
         :path "~/.emacs.d/elpa/skewer-mode-20180706.1807/skewer-mode.elc"
         :featurep t
         :version-var nil
         :version nil)))
```

**Comment by dickmao** [comment]
The failure to render the first time is documented in jupyter/notebook#3670 and #518 .
The failure to render the second time ("." instead of graph) is less understood.

Can you `M-x describe-char` on the "."?

The *display* text property needs to be present as follows:
```
There are text properties here:
  :pm-mode             fundamental-mode
  :pm-span             [Show]
  display              [Show]
  rear-nonsticky       (display)
```

**Comment by millejoh** [comment]
Maybe a silly question, but have you called `%matplotlib inline` before trying to create a plot?

**Comment by gahag** [comment]
Hello, thanks for the help!
Indeed, I was missing `%matplotlib inline`, and it fixed the first time rendering issue.
Now, the dot remains. Here is the output of `describe-char`:

```
             position: 140 of 158 (88%), column: 0
            character: . (displayed as .) (codepoint 46, #o56, #x2e)
              charset: ascii (ASCII (ISO646 IRV))
code point in charset: 0x2E
               script: latin
               syntax: . 	which means: punctuation
             category: .:Base, a:ASCII, l:Latin, r:Roman
             to input: type "C-x 8 RET 2e" or "C-x 8 RET FULL STOP"
          buffer code: #x2E
            file code: #x2E (encoded by coding system utf-8-unix)
              display: by this font (glyph code)
    xft:-ADBO-Source Code Pro-normal-normal-normal-*-13-*-*-*-m-0-iso10646-1 (#x391)

Character code properties: customize what to show
  name: FULL STOP
  old-name: PERIOD
  general-category: Po (Punctuation, Other)
  decomposition: (46) ('.')

There is an overlay here:
 From 140 to 142
  evaporate            t
  face                 ein:cell-output-area


There are text properties here:
  fontified            t
  keymap               [Show]
  rear-nonsticky       (display)
```

**Comment by gahag** [comment]
Okey, I solved it. Had to enable `ein:polymode`. Is this documented anywhere as a requirement for images? I've only seen it as a requirement for elpy, lsp, etc.

**Comment by dickmao** [comment]
It would be interesting to know if the image fails again if you turn off `ein:polymode`.

`ein:polymode` is strictly optional, and should not be required for basic image rendering.

When the image fails, and you `M-x ein:log-pop-to-all-buffer`, with any luck there should be a warning `Could not insert image: <explanation>`

**Comment by gahag** [comment]
Turning off `ein:polymode` brings back the problem. No messages about images in `ein:log-pop-to-all-buffer`. I guess I'll keep using polymode for now.
